PWGJE/Tasks/jetChCorr.cxx

```diff
@@ -0,0 +1,439 @@
+                registry.fill(HIST("hr2_ch_d_ft"), log10(fT));
+                registry.fill(HIST("hr2_ch_d_mult"), ch_mult);
+                registry.fill(HIST("hr2_ch_d_zg"), zg);
+                registry.fill(HIST("hr2_ch_d_rg"), log(rg));
+                registry.fill(HIST("hr2_ch_d_lu"), log(1 / rg), log(kt_p));
+              }
+            }
+            ////////  for 3rd marched split
+            if (nsd >= 2) {
+              if (ch_l == ch_nl) {
+                registry.fill(HIST("hr3_ch_s_pt"), ptJet);
+                registry.fill(HIST("hr3_ch_s_kt"), kt_p);
+                registry.fill(HIST("hr3_ch_s_z"), z);
+                registry.fill(HIST("hr3_ch_s_ft"), log10(fT));
+                registry.fill(HIST("hr3_ch_s_mult"), ch_mult);
+                registry.fill(HIST("hr3_ch_s_zg"), zg);
+                registry.fill(HIST("hr3_ch_s_rg"), log(rg));
+                registry.fill(HIST("hr3_ch_s_lu"), log(1 / rg), log(kt_p));
+              }
+              if (ch_l != ch_nl) {
+                registry.fill(HIST("hr3_ch_d_pt"), ptJet);
+                registry.fill(HIST("hr3_ch_d_kt"), kt_p);
+                registry.fill(HIST("hr3_ch_d_z"), z);
+                registry.fill(HIST("hr3_ch_d_ft"), log10(fT));
+                registry.fill(HIST("hr3_ch_d_mult"), ch_mult);
+                registry.fill(HIST("hr3_ch_d_zg"), zg);
+                registry.fill(HIST("hr3_ch_d_rg"), log(rg));
+                registry.fill(HIST("hr3_ch_d_lu"), log(1 / rg), log(kt_p));
+              }
+            }
+          }
+        }
+        nsd++;
+      }
+      daughterSubJet = parentSubJet1;
+    }
+    if constexpr (!isSubtracted && !isMCP) {
+      registry.fill(HIST("h2_jet_pt_jet_nsd"), jet.pt(), nsd);
+    }
+    if constexpr (!isSubtracted && isMCP) {
+      registry.fill(HIST("h2_jet_pt_part_jet_nsd_part"), jet.pt(), nsd);
+    }
+    if constexpr (isSubtracted && !isMCP) {
+      registry.fill(HIST("h2_jet_pt_jet_nsd_eventwiseconstituentsubtracted"), jet.pt(), nsd);
+    }
+    // outputTable(energyMotherVec, ptLeadingVec, ptSubLeadingVec, thetaVec);
+  }
+
+  template <bool isSubtracted, typename T, typename U, typename V>
+  void analyseCharged(T const& jet, U const& tracks, V& outputTable)
+  {
+    jetConstituents.clear();
+
+    // int leadpt =0;
+    // int n_leadpt =0;
+    int nn = 0;
+    ch_mult = 0; // jet.size();
+    ch_mult = jet.tracksIds().size();
+    // for (auto& jetConstituent : jet.template tracks_as<U>()) {
+    for (auto& jetConstituent : jet.template tracks_as<aod::JTracks>()) {
+      fastjetutilities::fillTracks(jetConstituent, jetConstituents, jetConstituent.globalIndex());
+
+      // int tr = jetConstituent.globalIndex();
+      // auto const& trackSel = tracks.iteratorAt(tr);
+
+      TVector3 p_leading(jetConstituent.px(), jetConstituent.py(), jetConstituent.pz());
+
+      if (nn == 0) {
+        // leadpt =  p_leading.Perp();
+        trackL = jetConstituent.globalIndex();
```

Comment:
i do not think its gaurunteed that the jet constituents are returned in a pT ordered way. You should explicitly loop over them and find the highest pT ones

Reply:
ordered included